### PR TITLE
fix nil time panic

### PR DESCRIPTION
**Pull request by nicolasparada**
Fix a panic due to `nil` `*time.Time`. This was happening in the delete agents command.

Also took the chance to update deps.

### CODE REVIEW

**Review by patrick-stephens — APPROVED** [review]

**Comment by patrick-stephens** [comment]
Should we add a unit test?

File: cmd/calyptia/top_project.go [hunk 1]
```diff
@@ -128,8 +128,8 @@ func (m ProjectModel) loadData(ctx context.Context, skipError bool) tea.Cmd {
 				for _, agent := range agents {
 					agent := agent
 					// skip metrics request if agent is offline
-					if agent.FirstMetricsAddedAt.IsZero() || agent.FirstMetricsAddedAt.After(metricsStart) ||
-						agent.LastMetricsAddedAt.IsZero() || agent.LastMetricsAddedAt.Before(metricsStart) {
+					if agent.FirstMetricsAddedAt == nil || agent.FirstMetricsAddedAt.IsZero() || agent.FirstMetricsAddedAt.After(metricsStart) ||
+						agent.LastMetricsAddedAt == nil || agent.LastMetricsAddedAt.IsZero() || agent.LastMetricsAddedAt.Before(metricsStart) {
```
Comment (gabrielbussolo) [inline cmd/calyptia/top_project.go:132]: This sentence is growing in new conditions (`||`) impairing readability maybe we can simplify it or divide for readability sake

**Review by gabrielbussolo — CHANGES_REQUESTED** [review]
Looks good for me, just i noticed we have growing the conditions to skip metrics, maybe we can break up into new functions or simplify for readability sake

File: cmd/calyptia/top_project.go (hunk 1 above)
Comment (patrick-stephens) [reply cmd/calyptia/top_project.go:132]: https://github.com/calyptia/cli/issues/297

**Pull request merged**